src/main/java/org/jabref/gui/importer/fetcher/WebSearchPaneViewModel.java

```diff
@@ -144,7 +146,20 @@ public void search() {
         }
 
         SearchBasedFetcher activeFetcher = getSelectedFetcher();
-        Callable<ParserResult> parserResultCallable = () -> new ParserResult(activeFetcher.performSearch(query));
+
+        ParserResult parserResult;
+        try {
+            parserResult = new ParserResult(activeFetcher.performSearch(query));
+        } catch (FetcherException e) {
+            if (e.getCause() != null && e.getCause().getCause() instanceof FetcherClientException clientException) {
```

Comment:
So we should stick with the original solution ?
what about the contents of dialogue do they need a change ?

Reply:
> So we should stick with the original solution ?

Yes. Please add a Java comment: "FetcherExcpetion's cause can be both IOException and ParseException", thus, there is no common parent.

> what about the contents of dialogue do they need a change ?

You are talking about the screenshot you provided above?

![image](https://github.com/user-attachments/assets/09403201-7ecb-4540-a7a4-726e682fb335)

Exceptions are details. With the issue, we want to make the http response more prominent. Please move it up and also add the http body to the output.

Please create a new screenshot (update your PR description) so that we can review more easily.